pkg/sm/security_builder.go

```diff
@@ -0,0 +1,273 @@
+package sm
+
+import (
+	"fmt"
+
+	"github.com/Peripli/service-manager/api/filters"
+	secFilters "github.com/Peripli/service-manager/pkg/security/filters"
+
+	httpsec "github.com/Peripli/service-manager/pkg/security/http"
+	"github.com/Peripli/service-manager/pkg/security/http/authn"
+	"github.com/Peripli/service-manager/pkg/security/http/authz"
+	"github.com/Peripli/service-manager/pkg/web"
+)
+
+type securityBuilder struct {
```

Comment:
It's great how you've documented each method below, but can we also have some overview go doc as to how to use the securityBuilder? In particular for me it's missing the part where we state that `Required()` and `Optional()` are soft of like finisher methods.

Maybe something like:

```go
// securityBuilder provides means by which authentication and authorization filters
// can be constructed and attached in a builder-pattern style through the use of methods such as:
// Path(...), Method(...), WithAuthentication(...), WithAuthorization(...) and more.
// A key part of the builder is that once you've chained all desired authentication
// and authorization settings for a specific API (a set of path and method parameters)
// you have to use one of the provided finisher methods - Required() or Optional().
// These finisher methods will ensure that the appropriate authentication/authorization filter is constructed for 
// the desired path and methods. A finisher method also ensures a "clean slate" in terms of authorization
// so that you continue chaining and constructing new authorization filters.
type securityBuilder struct {
...
}
```